libraries/ESP8266MQTTClient/src/URIParser.h

```diff
@@ -0,0 +1,28 @@
+#ifndef URIParser_h_
+#define URIParser_h_
+typedef struct{
+	char *scheme;               /* mandatory */
+	char *host;                 /* mandatory */
+	char *port;                 /* optional */
+	char *path;                 /* optional */
+	char *query;                /* optional */
+	char *fragment;             /* optional */
+	char *username;             /* optional */
+	char *password;             /* optional */
+} parsed_url;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+    /*
+     * Declaration of function prototypes
+     */
+    parsed_url *parse_uri(const char *);
+    void http_parsed_url_free(parsed_url *);
```

Comment:
maybe ```free_parsed_url``` sounds a bit better? no need to get "http" into the mix since it's not mentioned anywhere and the url can be of other scheme (```file://test.txt```). Putting what you are freeing after the word "free" makes it sound more english :)